CRM/PivotData/AbstractData.php

```diff
@@ -104,13 +111,72 @@ public function __construct($name, $apiEntityName = NULL) {
     $this->name = $name;
     $this->apiEntityName = $apiEntityName ? $apiEntityName : $name;
 
+    $this->loadLabelCustomizations();
     $dateFields = $this->getDateFields();
 
     foreach ($dateFields as $field) {
       $this->additionalHeaderFields[$field . ' (' . ts('per month') . ')'] = '';
     }
   }
 
+  /**
+   * Loads customized labels from json configuration files.
+   */
+  private function loadLabelCustomizations() {
+    $className = strtr(get_class($this), array('CRM_PivotData_' => ''));
+    $customLabelsFileName = $this->getDefaultResourcesPath() . 'custom_labels/' . $className . '.json';
+
+    if(file_exists($customLabelsFileName)) {
+      $jsonData = file_get_contents($customLabelsFileName);
+      $this->customLabels = json_decode($jsonData, true);
+    }
+  }
+
+  /**
+   * Returns default resources path of the extension.
+   *
+   * @return string
+   */
+  private function getDefaultResourcesPath() {
+    return realpath(__DIR__ . '/../../') . '/resources/';
```

Comment:
you can use : 

```php
$extensionDirectory = CRM_Core_Resources::singleton()->getPath('EXTENSION_KEY');
```

to get the extension directly.

I would even remove this method and move this line (after changing it to my suggestion above) to the method above.